### PR TITLE
F test on hw

### PR DESCRIPTION
**Pull request by toliver**

### CODE REVIEW

File: sr_self_test/src/test_joint_movement.cpp [hunk 1]
```diff
@@ -66,10 +95,53 @@
...
+  }
+
   void TestJointMovement::mse_cb_(const std_msgs::Float64::ConstPtr& msg)
   {
     mse = msg->data;
 
     //unsubscribe after receiving the message
   }
+
+  std::string TestJointMovement::get_ROS_topic_type(std::string topic_name)
```
Comment (ugocupcic) [inline sr_self_test/src/test_joint_movement.cpp:112]: yuck ! Using popen to call rostopic to get the topic type.... not cool. I hope we can use getSubscribedTopics from the node handle instead (https://code.ros.org/trac/ros/ticket/2693)

File: sr_self_test/src/sensor_noise_test.cpp [hunk 2]
```diff
@@ -65,7 +73,9 @@
         }
 
         double noise = max - min;
-        if( noise > MAX_NOISE_CONST_ || noise == 0.0 )
+        ROS_INFO("Joint %s Pos min: %lf  max: %lf  noise: %lf", joint_states->first,  min , max , noise );
```
Comment (ugocupcic) [inline sr_self_test/src/sensor_noise_test.cpp:76]: This one doesn't compile on my computer:

error: cannot pass objects of non-trivially-copyable type ‘const struct std::basic_string<char>’ through ‘...’
warning: format ‘%s’ expects argument of type ‘char*’, but argument 8 has type ‘const std::basic_string<char>’ [-Wformat]

File: sr_self_test/src/test_joint_movement.cpp (hunk 1 above)
Comment (ugocupcic) [inline sr_self_test/src/test_joint_movement.cpp:112]: :( getPublishedTopics doesn't seem to exist anymore unfortunately.

File: sr_self_test/src/sensor_noise_test.cpp (hunk 2 above)
Comment (toliver) [inline sr_self_test/src/sensor_noise_test.cpp:76]: Sorry about that. It should be joint_states->first.c_str()
I've fixed that in the other merge request.
(I know it shouldn't have been merged with this error :(

File: sr_self_test/src/test_joint_movement.cpp (hunk 1 above)
Comment (toliver) [inline sr_self_test/src/test_joint_movement.cpp:112]: I know it's far from optimal, but we only use it very seldom. Once for every joint movement test.

Oddly enough ROS doesn't have a stable API to get this information, and for C++ not even an unstable API.

File: sr_self_test/src/test_joint_movement.cpp [hunk 3]
```diff
@@ -66,10 +95,53 @@
...
+    
+    FILE *myfile;
+    std::string cmd;
+
+    cmd = "rostopic type " + topic_name;
+
+    // make sure to popen and it succeeds
+    if(!(myfile = popen(cmd.c_str(), "r")))
+    {
+      ROS_ERROR_STREAM("Command failed: " << cmd);
```
Comment (toliver) [inline sr_self_test/src/test_joint_movement.cpp:125]: By the way, I should return a string here. (It hasn't happened yet).

File: sr_self_test/src/test_joint_movement.cpp (hunk 1 above)
Comment (ugocupcic) [inline sr_self_test/src/test_joint_movement.cpp:112]: Yeah I've been looking for one for some time now... very weird.

On 29 May 2013 12:26, toliver notifications@github.com wrote:

> In sr_self_test/src/test_joint_movement.cpp:
> 
> >    void TestJointMovement::mse_cb_(const std_msgs::Float64::ConstPtr& msg)
> >    {
> >      mse = msg->data;
> > 
> > ```
> >  //unsubscribe after receiving the message
> > ```
> > 
> >    }
> > +
> > -  std::string TestJointMovement::get_ROS_topic_type(std::string topic_name)
> 
> I know it's far from optimal, but we only use it very seldom. Once for
> every joint movement test.
> 
> Oddly enough ROS doesn't have a stable API to get this information, and
> for C++ not even an unstable API.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/shadow-robot/sr-ros-interface/pull/39/files#r4431566
> .

## 

Ugo Cupcic | Shadow Robot Company | ugo@shadowrobot.com
Senior Software Engineer | 251 Liverpool Road |
need a Hand? | London N1 1LX | +44 20 7700 2487
http://www.shadowrobot.com/hand/ @shadowrobot

**Pull request merged**